### ISSUE TITLE
Need to reset terminal colors to the original setting when outputting error to the terminal.

### ISSUE THREAD

**Issue by TierraDelFuego** [okamstudio/godot#2566]
^[[1;31;40mERROR: cleanup: ^[[1;37;40mObjectDB Instances still exist!
^[[0;31;40m   At: core/object.cpp:1853.^[[0;0;37m

This leaves the terminal color, i.e. the prompt white instead of red or green or blue or any color you had for your prompt before Godot wrote an error message to the terminal.

**Comment by TierraDelFuego** [comment]
I made the change in my repo, I had some problems with the user.name and so there are three commits to that file from my attempt to fix that. That is why I didn't send a pull request.

https://github.com/TierraDelFuego/godot/blob/master/drivers/unix/os_unix.cpp

**Comment by slapin** [comment]
I think an additional option to remove ANSI colors at all at compile time
should be added for completeness

**Comment by akien-mga** [comment]
Good point, I hadn't noticed this as I use a white font in my terminal.

@TierraDelFuego Is the fix the addition of ```\E[0m```? Then I guess ```\E[0;0;37m``` can be removed too. I'll try that in my current PR about error output (#2737).

**Comment by akien-mga** [comment]
I can confirm the bug in a terminal with a white background and dark fonts, e.g. here gnome-terminal:
![gt-colour](https://cloud.githubusercontent.com/assets/4701338/11056236/009585e6-8780-11e5-99eb-69a07259af94.jpg)

**Comment by akien-mga** [comment]
The issue should now be fixed once my PR is merged. Then there is still the issue of a black background forced on error messages, but if we want to address that we'd likely have to find font colours which are constrasted enough on any terminal background colour. This could be done in a second step.

**Comment by slapin** [comment]
I think terminal colors should be possible to be disabled from command line

**Comment by akien-mga** [comment]
> I think terminal colors should be possible to be disabled from command line

That's a good idea, but also a separate issue. I'll open one.
